File: src/pocketcloud/cloud/network/packet/impl/normal/LibrarySyncPacket.php
<?php

namespace pocketcloud\cloud\network\packet\impl\normal;

use pocketcloud\cloud\library\Library;
use pocketcloud\cloud\library\LibraryManager;
use pocketcloud\cloud\network\client\ServerClient;
use pocketcloud\cloud\network\packet\CloudPacket;
use pocketcloud\cloud\network\packet\data\PacketData;

class LibrarySyncPacket extends CloudPacket {

    private array $data = [];

    public function __construct() {
        foreach (array_filter(LibraryManager::getInstance()->getAll(), fn(Library $library) => $library->isCloudBridgeOnly()) as $lib) {
            $this->data[] = [
                "name" => $lib->getName(),
                "path" => $lib->getUnzipLocation()
            ];
        }
    }

    public function encodePayload(PacketData $packetData): void {
        $packetData->write($this->data);
    }

    public function decodePayload(PacketData $packetData): void {
        $this->data = $packetData->readArray();
    }

    public function getData(): array {
        return $this->data;
    }

    public function handle(ServerClient $client): void {}

    public static function create(): self {
        return new self();
    }
}

File: src/main/java/de/pocketcloud/cloud/network/packet/impl/LibrarySyncPacket.java
package de.pocketcloud.cloud.network.packet.impl;

import de.pocketcloud.cloud.server.CloudServer;
import de.pocketcloud.cloud.server.library.Library;
import de.pocketcloud.cloud.server.library.LibraryManager;
import de.pocketcloud.cloud.network.client.ServerClient;
import de.pocketcloud.cloud.network.packet.ClientboundPacket;
import de.pocketcloud.cloud.network.packet.CloudPacket;
import de.pocketcloud.cloud.network.packet.data.PacketData;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.jetbrains.annotations.NotNull;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;

@NoArgsConstructor
@Getter
public final class LibrarySyncPacket extends CloudPacket implements ClientboundPacket {

    private List<LinkedHashMap<String, String>> data;

    public LibrarySyncPacket(List<LinkedHashMap<String, String>> data) {
        this.data = data != null ? data : List.of();
    }

    @Override
    public void handle(@NotNull ServerClient client) {}

    @Override
    public void encodePayload(PacketData packetData) {
        packetData.writeAll(data);
    }

    @Override
    public void decodePayload(PacketData packetData) {}

    public static LibrarySyncPacket create(List<LinkedHashMap<String, String>> data) {
        return new LibrarySyncPacket(data);
    }

    public static LibrarySyncPacket fromLibraries(CloudServer server) {
        List<LinkedHashMap<String, String>> data = new ArrayList<>();
        for (Library lib : LibraryManager.instance().getAll().values()) {
            if (!lib.isAvailableFor(server.template().serverSoftware())) continue;
            LinkedHashMap<String, String> libData = new LinkedHashMap<>();
            libData.put("name", lib.name());
            libData.put("path", lib.directoryPath().toAbsolutePath().toString());
            libData.put("namespacePrefix", lib.namespacePrefix());
            libData.put("namespaceFolder", lib.namespaceFolder());
            data.add(libData);
        }
        return new LibrarySyncPacket(data);
    }
}